.github/workflows/merge-check.yml

```diff
@@ -46,21 +46,21 @@ jobs:
       - name: Run Gradle tasks
         run: ./gradlew build 
 
-      - name: Upload 'Test Report' artifact
-        uses: actions/upload-artifact@a8a3f3ad30e3422c9c7b888a15615d19a852ae32 # v3
+      - name: Upload 'Test Report (${{ matrix.os }})' artifact
+        uses: actions/upload-artifact@26f96dfa697d77e81fd5907df203aa23a56210a8 # v4
         if: success() || failure()
         with:
-          name: Test Report
+          name: Test Report (${{ matrix.os }})
```

Comment:
It seems previously the reports were totally overwriting each other, obviously they should be the same, but we wouldn't have known if they weren't.